gdc_ng_models/models/studyrule.py

```diff
@@ -0,0 +1,137 @@
+"""Data models to describe studies.
+
+Studies will be used for the single-study controlled-access feature in gdcapi.
+These models provide a way to associate programs and projects to studies.
+"""
+from sqlalchemy import orm
+from sqlalchemy.ext import declarative
+from sqlalchemy.sql import schema, sqltypes
+
+from gdc_ng_models.models import audit
+
+Base = declarative.declarative_base()
+
+
+class StudyRule(Base, audit.AuditColumnsMixin):
+    """A study rule for use with single-study controlled-access.
+
+    Attributes:
+        id: A unique identifier for the study rule.
+        name: A unique, informational name for the study rule.
+        created_date: The date and time when the record is created.
+        updated_date: The date and time when the record is updated.
+    """
+
+    __tablename__ = "study_rule"
+    id_seq = schema.Sequence(name="study_rule_id_seq", metadata=Base.metadata)
+    id = schema.Column(sqltypes.Integer, nullable=False, server_default=id_seq.next_value())
+    name = schema.Column(sqltypes.Text, nullable=False)
+
+    __table_args__ = (
+        schema.PrimaryKeyConstraint("id", name="study_rule_pk"),
+        schema.Index("study_rule_name_idx", "name", unique=True),
+    )
+
+    whole_programs = orm.relationship("StudyRuleProgram", lazy="joined")
+    partial_programs = orm.relationship("StudyRuleProgramProject", lazy="joined")
```

Comment:
The stub model in gdcapi currently assumes the relationship will be more like study -> program -> project, but I can see advantages to this approach (and this is admittedly more to-spec) and am good with reworking the API logic to expect this.

Reply:
I assumed we'd have a function that transforms the data model to the stub models, especially given the comment:
```
    Differs slightly from :class:`StudyProgram` in that all projects must be listed
    even if the rule covers the whole program...
```
This would require getting all of the released projects for a whole program, which is not modeled in the study rule data models.

Reply:
Yeah, I think it's reasonable to assume some transformation/mapping will be done. We might be able to avoid that by changing assumptions in other parts of the code, but in any case, I don't think that should affect how this is modeled.